### ISSUE TITLE
Consumer Service register error when with scope lifetime

### ISSUE THREAD

**Issue by sampsonye** [dotnetcore/CAP#1204]
如果消费端服务生命周期是Scope会报错
![image](https://user-images.githubusercontent.com/5895634/189293489-b179000c-883e-430f-83a8-acbe41bfe805.png)

**Comment by yang-xiaodong** [comment]
How to reproduce?

**Comment by sampsonye** [comment]
It's happend only if the class register itself as Scope service(Factory Mode)
```
public class DemoService :  ICapSubscribe
{
    public virtual string Add(int a, int b)
    {
        throw new NotImplementedException();
    }

    [CapSubscribe("demo")]
    public Task DemoSubscribe(DateTime time)
    {
        return Task.CompletedTask;
    }
}
```
Register :
```
builder.Services.AddScoped<DemoService>(sp =>
{
    var d = new DemoService();
    return d;
});
```

reproduce Demo like :https://github.com/sampsonye/cap-issus-1204

**Comment by sampsonye** [comment]
any questions? @yang-xiaodong

**Comment by yang-xiaodong** [comment]
Fixed in v6.2.0
